src/GitVersion.Core.Tests/IntegrationTests/CreatingAFeatureBranchFromAReleaseBranchScenario.cs

```diff
@@ -0,0 +1,701 @@
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-just-a-test.1+2", configuration);
+
+        fixture.Checkout("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+1", configuration);
+
+        fixture.MergeNoFF("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+3", configuration);
+
+        fixture.Repository.Branches.Remove("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+3", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+4", configuration);
+
+        fixture.Repository.DumpGraph();
+    }
+
+    [Test]
+    public void ShouldTreatTheMergeFromReleaseToDevelopLikeTheReleaseBranchHasNeverBeenExistingWhenReleaseHasBeenCanceled()
+    {
+        // *809eaa7 in the future(HEAD -> develop)
+        // *46e2cb8 in the future
+        // |\  
+        // | *08bd8ff in the future
+        // | *9b741de in the future
+        // * | 13206fd in the future
+        // |/
+        // *9dc9b22 in the future
+        // *f708abd in the future(main)
+
+        var configuration = TestConfigurationBuilder.New.Build();
+
+        using var fixture = new EmptyRepositoryFixture();
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.0.1+1", configuration);
+
+        fixture.BranchTo("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.2", configuration);
+
+        fixture.BranchTo("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+2", configuration);
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configuration);
+
+        fixture.MergeNoFF("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.4", configuration);
+
+        fixture.Repository.Branches.Remove("release/1.0.0");
+
+        // ❌ expected: "0.1.0+6" because the release has been canceled and should be treated like it was never existing
+        fixture.AssertFullSemver("1.1.0-alpha.4", configuration);
```

Comment:
Actually this is a good question. In my opinion the should be configurable in future. I already discussed this behavior in the issue: https://github.com/GitTools/GitVersion/issues/3052#issuecomment-1246245489

In my opinion we should introduce a new filed track-merge-message:

> One thing may be an idea to separate this and make it configurable. This would be easier to implement and better for the understanding. Concrete my proposal is to separate this two features and introducing a new configuration property like track-merge-message.

If you do this you can change the configuration in develop branch not detecting this changes.

Reply:
Okay, I see. Yes, `track-merge-message` makes sense. With that introduced, perhaps we should group all tracking properties into a parent `tracking` property, like so:

```yaml
branches:
  develop:
    tracking:
      merge-target: true
      merge-message: true
      release-branches: true
```

Or perhaps we should make it possible to configure which strategies each branch can use directly?

```yaml
branches:
  develop:
    strategies:
      - BranchName
      - Fallback
      - MergeMessage
      - NextVersion
      - ReleaseBranches
      - TaggedCommit
```

Reply:
You mean to make it configurable which IVersionStrategy implementation should be used for what branch!? In my opinion we should not introduce a dependency on that layer. It should be indirect controlled with dedicated properties which can be defined on each branch and maybe inherited dependent on the IncrementStrategy.

That means I would prefer the first proposal with dedicated properties. For example you could have a IVersionStrategy implementation with more then one property. If you want to encapsulate it in a sub class never mind. At the end it's the same.

We could think of to introduce a mechanism to make the IVersionStrategy like a provider the user can extend on a global level. But therefore we need to make the BranchConfig class generic with e.g. JObject. But for now this generic provider thing is out of scope IMO.

Reply:
Done